Docs/src/ogre-shadows.md

```diff
@@ -397,15 +400,386 @@ Non-PCF block. Note that after doing this, the uSTexWidth and
 uSTexHeight variables are likely to be optimized away and so you should
 uncomment these variables in the materials script as well.
 
-The following shows how to activate plane optimal shadow mapping given
-some pointer to a MovablePlane and a pointer to a light.
+## Debugging Shadows
+Since shadows are a difficult subject, so it is a good idea to have the Shadow Map 
+projected on a Mini-Screen where it is possible to see how the Depth Caster is performing.
+
+Material definition: `shadow_debug.material`
+```
+material ShadowDebug
+{
+	technique
+    {
+        pass 
+        {
+            vertex_program_ref DebugVP {}
+            fragment_program_ref DebugFP {}
+
+            texture_unit ShadowMap
+            {
+                tex_address_mode clamp
+                filtering none
+				content_type shadow
+            }
+        }
+    }
+}
+```
+
+Program definition: `shadow_debug.program`
+```
+vertex_program DebugVP glsl
+{
+	source "shadow_debug.vs"
+}
+
+fragment_program DebugFP glsl
+{
+	source "shadow_debug.fs"
+
+	default_params
+    {
+		param_named shadowMap int 0
+    }
+}
+```
+
+Vertex shader: `shadow_debug.vs`:
+```glsl
+#version 330 core
+
+layout (location = 0) in vec4 vertex;
+layout (location = 8) in vec2 uv0;
+
+out vec2 TexCoords;
+
+void main()
+{
+    gl_Position = vec4(vertex.x, vertex.y, 0.0, 1.0); 
```

Comment:
this does not account for vertical texture flipping on GL. With ShadowMaps you will not notice, but still.
See the StdQuad_Tex2_vp.glsl sample